Objects/methodobject.c

```diff
@@ -60,11 +70,44 @@ PyCFunction_NewEx(PyMethodDef *ml, PyObject *self, PyObject *module)
             return NULL;
     }
 
-    PyCFunctionObject *op =
-        PyObject_GC_New(PyCFunctionObject, &PyCFunction_Type);
-    if (op == NULL) {
-        return NULL;
+    PyCFunctionObject *op = NULL;
+
+    if (ml->ml_flags & METH_METHOD) {
+        PyCMethodObject *om;
+        if (ml->ml_flags & (METH_NOARGS | METH_O | METH_CLASS | METH_STATIC)) {
+            PyErr_SetString(PyExc_SystemError,
+                            "METH_METHOD cannot be used with METH_NOARGS, "
+                            "METH_O, METH_CLASS, nor METH_STATIC");
+            return NULL;
+        }
```

Comment:
This looks like it would allow `(METH_VARARGS | METH_METHOD)` or `METH_FASTCALL | METH_METHOD`